bitcoin/test/fuzz/miniscript_decode.cpp

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <core_io.h>
+#include <hash.h>
+#include <key.h>
+#include <script/miniscript.h>
+#include <script/script.h>
+#include <span.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <util/strencodings.h>
+
+#include <optional>
+
+
+struct Satisfier: BaseSignatureChecker {
+    typedef CPubKey Key;
+
+    // Precomputed public keys, and a dummy signature for each of them.
```

Comment:
This uses a single object as holder of the precomputed data, converter ctx, satisfier ctx, and checker ctx. I guess the approach in #94 is cleaner..